### PR TITLE
add CI check to test if a crate compiles to wasm

### PR DESCRIPTION
**Pull request by clangenb**
The benchmarks don't compile in the node currently, with the proper CI check this could not have been overlooked. This PR adds a CI check that tests for each crate individually if it compiles to wasm, and it enabled the `runtime-benchmarks` feature if a crate has that feature.

Basically this PR ensures that all pallets work with:
* cargo test -p "crate" --features runtime-benchmarks 
* clear && cargo check -p "crate" --no-default-features --features runtime-benchmarks --target=wasm32-unknown-unknown  

Both tests are executed in the CI already

### CODE REVIEW

File: pallets/ajuna-affiliates/src/benchmarking.rs [hunk 1]
```diff
@@ -14,33 +14,10 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-#![cfg(feature = "runtime-benchmarks")]
-#![cfg_attr(not(feature = "std"), no_std)]
-
-use crate::{
-	mock::{
```
Comment (clangenb) [inline pallets/ajuna-affiliates/src/benchmarking.rs:21]: No mock imports here!

File: pallets/ajuna-affiliates/src/benchmarking.rs [hunk 2]
```diff
@@ -136,21 +123,8 @@ benchmarks_instance_pallet! {
...
-	let mut ext = sp_io::TestExternalities::new(t);
-	ext.execute_with(|| System::set_block_number(1));
-	ext.execute_with(|| {
-		let acc_1 = account::<Test, ()>(ACC_1);
-		MockAccountManager::set_organizer(acc_1);
-		MockAccountManager::try_add_to_whitelist(&AffiliateWhitelistKey::get(), &acc_1)
-			.expect("Should add to whitelist");
-	});
-	ext
-}
```
Comment (clangenb) [inline pallets/ajuna-affiliates/src/benchmarking.rs:156]: The setup must not contain anything besides the default externalities. All custom state must be done in the benchmark setup function.

File: pallets/ajuna-affiliates/src/lib.rs [hunk 3]
```diff
@@ -18,7 +18,7 @@
 
 pub use pallet::*;
 
-#[cfg(any(test, feature = "runtime-benchmarks"))]
+#[cfg(any(test))]
```
Comment (clangenb) [inline pallets/ajuna-affiliates/src/lib.rs]: It is best to not enable this with the runtime-features flag to ensure that we don't use stuff in the benchmarks we are not supposed to.

File: primitives/src/account_manager.rs [hunk 4]
```diff
@@ -27,4 +27,13 @@ pub trait AccountManager {
...
 	fn is_whitelisted_for(identifier: &WhitelistKey, account: &Self::AccountId) -> bool;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn set_organizer(account: Self::AccountId);
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn try_add_to_whitelist(
+		identifier: &WhitelistKey,
+		account: Self::AccountId,
+	) -> Result<(), DispatchError>;
```
Comment (clangenb) [inline primitives/src/account_manager.rs:38]: We need to have this in the trait definition if we need to use this in the benchmarks too.

**Comment by DidacSF** [comment]
I've finished adapting the rest of the code @clangenb I believe the PR is now ready for review!

File: pallets/ajuna-awesome-avatars/src/impls/account_manager.rs [hunk 5]
```diff
@@ -29,4 +29,17 @@ impl<T: Config> AccountManager for Pallet<T> {
...
+		todo!()
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn try_add_to_whitelist(
+		_identifier: &WhitelistKey,
+		_account: Self::AccountId,
+	) -> Result<(), DispatchError> {
+		todo!()
+	}
```
Comment (clangenb) [inline pallets/ajuna-awesome-avatars/src/impls/account_manager.rs]: Can we please implement this? Otherwise, the benchmark will fail at runtime in the node, as we do use these functions.

File: pallets/ajuna-nft-transfer/src/benchmarking.rs [hunk 6]
```diff
@@ -64,7 +50,7 @@ fn create_service_account_and_prepare_avatar<T: Config>(
 
 fn enable_fee_payment<T: Config>(player: &T::AccountId) {
 	let prepare_fee = 100_000_000_000_000u128;
-	CurrencyOf::<T>::make_free_balance_be(player, prepare_fee.saturated_into());
+	T::BenchmarkHelper::set_account_balance(player, prepare_fee.saturated_into());
```
Comment (clangenb) [inline pallets/ajuna-nft-transfer/src/benchmarking.rs]: Why did you change this? I think this is not a good pattern. It forces us to implement something in the runtime that is unnecessary.

File: pallets/ajuna-nft-transfer/src/benchmarking.rs [hunk 7]
```diff
@@ -15,31 +15,17 @@
 // along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #![cfg(feature = "runtime-benchmarks")]
```
Comment (clangenb) [inline pallets/ajuna-nft-transfer/src/benchmarking.rs]: I recommend removing this feature flag. The whole module is already feature flagged in the lib.rs.

File: pallets/ajuna-nft-transfer/src/lib.rs [hunk 8]
```diff
@@ -60,6 +60,8 @@ pub mod pallet {
 	#[cfg(feature = "runtime-benchmarks")]
 	pub trait BenchmarkHelper<AccountId, ItemId> {
 		fn create_items(owner: AccountId, count: u32) -> sp_std::vec::Vec<ItemId>;
+
+		fn set_account_balance(account: &AccountId, balance: u32);
```
Comment (clangenb) [inline pallets/ajuna-nft-transfer/src/lib.rs]: As said before, I don't think this is a good idea.

File: sage/pallet/src/mock.rs [hunk 9]
```diff
@@ -480,3 +480,43 @@ impl ExtBuilder {
 		ext
 	}
 }
+
+pub fn new_test_ext() -> sp_io::TestExternalities {
+	ExtBuilder::default().build()
+}
+
+/*pub fn new_benchmark_ext() -> sp_io::TestExternalities {
```
Comment (clangenb) [inline sage/pallet/src/mock.rs]: I guess this was forgotten :)

File: sage/pallet/src/trait_impls/account_manager.rs [hunk 10]
```diff
@@ -28,4 +28,18 @@ impl<T: Config<I>, I: 'static> AccountManager for Pallet<T, I> {
...
+		Organizer::<T, I>::put(&account);
+		Self::deposit_event(Event::OrganizerSet { organizer: account });
+	}
+
+	#[cfg(feature = "runtime-benchmarks")]
+	fn try_add_to_whitelist(
+		_identifier: &WhitelistKey,
+		_account: Self::AccountId,
+	) -> Result<(), DispatchError> {
+		unimplemented!()
```
Comment (clangenb) [inline sage/pallet/src/trait_impls/account_manager.rs:43]: #tomorrowmeproblem XD.

Fine to leave it for now IMO.

File: pallets/ajuna-nft-transfer/src/benchmarking.rs (hunk 6 above)
Comment (DidacSF) [reply pallets/ajuna-nft-transfer/src/benchmarking.rs]: Because of this:

```
error[E0277]: the trait bound `T: pallet_nfts::Config` is not satisfied
  --> pallets/ajuna-nft-transfer/src/benchmarking.rs:55:2
   |
55 |     CurrencyOf::<T>::make_free_balance_be(player, prepare_fee.saturated_into());
   |     ^^^^^^^^^^^^^^^ the trait `pallet_nfts::Config` is not implemented for `T`
   |
help: consider further restricting this bound
   |
53 | fn enable_fee_payment<T: Config + pallet_nfts::Config>(player: &T::AccountId) {
   |                                 +++++++++++++++++++++
```

File: pallets/ajuna-nft-transfer/src/lib.rs (hunk 8 above)
Comment (DidacSF) [reply pallets/ajuna-nft-transfer/src/lib.rs]: See reason in other comment

File: pallets/ajuna-nft-transfer/src/benchmarking.rs (hunk 6 above)
Comment (clangenb) [reply pallets/ajuna-nft-transfer/src/benchmarking.rs]: Ok, this is bad design by us. The pallet needs to transfer funds, but it doesn't expose what is used to manage funds. I prefer though to add the constraint `pallet_balances::Config` here. As this is the assumption that the nft-transfer pallet has currently, and it is fullfilled by our runtime.

In the future, we probably want to have a `set_balance` feature in our fee-handler, gated behind a runtime-benchmarks feature ;)

Comment (DidacSF) [reply pallets/ajuna-nft-transfer/src/benchmarking.rs]: Okay fixed it using `where_clause` in the benchmarks

**Review by clangenb — COMMENTED** [review]
Ok, wonderful. Even better to depend on the  `nfts-pallets` than the balances pallet! Thanks for all the fixes, LGTM now!
